adlfs/transfer.py

```diff
@@ -0,0 +1,368 @@
+# -*- coding: utf-8 -*-
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+High performance multi-threaded module to up/download
+
+Calls method in `core` with thread pool executor to ensure the network
+is used to its maximum throughput.
+
+Only implements upload and download of (massive) files and directory trees.
```

Comment:
This docstring was copied from the `multithread` module and doesn't make sense here. I've updated it to be more descriptive.